### DISCUSSION TITLE
Exploring values with a different approach than grid tool

### DISCUSSION THREAD

**Discussion by Shadetail** [Stability-AI/StableSwarmUI#203] [Ideas]
I very often want to explore image creativity values between 0.3 and 0.7 for example, and when I find the ideal one, then I want to make a lot of images using that value.

Grid Generator tool seems to be the intended way of doing this, but same as in automatic1111, or even a bit more, I'm finding it cumbersome to use. I don't actually want any grids.

It's actually the nmkd UI that had the best solution to this, where each slider has a text input field next to it that can be used to specify the range of values to explore and the number of images to generate along that range (which then overrides the slider). It's really just replacing the slider with 3 numbers, it should be that simple, and that easy and fast to use. I currently find myself manually changing values and queueing many images as it's usually faster than using the Grid Generator, setting it up, figuring out where the images ended, moving them where they should be, cleaning up after it,...

**Comment by mcmonkey4eva** [comment]
There's actually a related feature here I've been planning, that I went ahead and added now which should clean up at least some of your needs there.

The Grid Generator tool now has a selector between Web Page, Just Images, or Grid Image.
If you use Just Images, it'll just output all the combinatoric images to the main image folder, not make the "end grid" at all.
if you use Grid Image, it's like Just Images, but it makes a single-image grid containing all of them (like that old auto webui built in xy plot script thingy did)

Also fwiw in the grid generator tool, you can just eg set Image Creativity to `0.3, 0.4, .., 0.7` ... the `..` is interpreted by the system to mean `0.5, 0.6` should fill in the space automatically.
This is more useful with larger sets, eg if you do Seed as `1, 2, .., 50` that will generate all 50 seeds.
(Two values before to set the start and step, one value after to set the endpoint).
Can even mix it in to a larger list, eg set your CFG Scale to `1, 3, 5, 5.5, .., 9, 11, .., 19` will fill to `1, 3, 5, 5.5, 6, 6.5, 7, 7.5, 8, 8.5, 9, 11, 13, 15, 17, 19`

Integrating more directly to the primary parameters list would require a larger UI rework

**Reply by Shadetail** [reply]
That's awesome! Thank you so much!
